### PR TITLE
[onert] Add Shape inference for ReduceSum

### PR DESCRIPTION
**Pull request by YiHyunJin**
This PR adds shape inference for ReduceSum op.

Signed-off-by: YiHyunjin <hj0412.yi@samsung.com>

### CODE REVIEW

**Comment by glistening** [comment]
@nnfw-bot test cross-aarch64-runtime-test-release

**Comment by glistening** [comment]
@YiHyunJin Could you please add test case for `keepDim=false`?

**Comment by glistening** [comment]
@nnfw-bot test cross-runtime-test-release

**Comment by glistening** [comment]
@nnfw-bot test cross-runtime-test-release

**Comment by glistening** [comment]
@YiHyunJin Please take a look the error:

```
[2020-06-05T06:35:38.287Z] [ RUN      ] GeneratedTests.reduce_sum_4D_float_reducing_HW_nnfw
[2020-06-05T06:35:38.287Z] /opt/jenkins_slave/workspace/nnfw/master/pr-cross-runtime-test-pipeline-release@2/tests/nnapi/src/TestGenerated.cpp:153: Failure
[2020-06-05T06:35:38.287Z]       Expected: Result::NO_ERROR
[2020-06-05T06:35:38.287Z]       Which is: 4-byte object <00-00 00-00>
[2020-06-05T06:35:38.287Z] To be equal to: execution.setOutput(idx, buffer, s)
[2020-06-05T06:35:38.287Z]       Which is: 4-byte object <04-00 00-00>
[2020-06-05T06:35:38.287Z] Google Test trace:
```

File: runtime/onert/core/src/util/shapeinf/ReduceSum.cc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    // Calculates size of reducing axis.
+    int num_reduce_axis = num_axis;
+    for (int i = 0; i < num_axis; ++i)
+    {
+      int current = axes[i];
+      if (current < 0)
+      {
+        current += input_num_dims;
+      }
+      assert(0 <= current && current < rank);
```
Comment (intom) [inline runtime/onert/core/src/util/shapeinf/ReduceSum.cc:70]: when I pull this PR, I meet the error. 'rank' was not declared in this scope.
what do you mean this "rank"?

Comment (YiHyunJin) [reply runtime/onert/core/src/util/shapeinf/ReduceSum.cc]: Thank you. I modified it.

**Comment by YiHyunJin** [comment]
@nnfw-bot test all

**Pull request merged**